src/coreclr/jit/gcinfo.cpp

```diff
@@ -254,6 +254,15 @@ GCInfo::WriteBarrierForm GCInfo::gcIsWriteBarrierCandidate(GenTreeStoreInd* stor
 
     if (store->Data()->IsIconHandle(GTF_ICON_OBJ_HDL))
     {
+#ifndef TARGET_XARCH
+        const ssize_t handle = store->Data()->AsIntCon()->IconValue();
+        if (!compiler->info.compCompHnd->isObjectImmutable(reinterpret_cast<void*>(handle)))
+        {
+            // See https://github.com/dotnet/runtime/pull/76135#issuecomment-1257258310
+            store->gtFlags |= GTF_IND_VOLATILE;
```

Comment:
This will work, but later it may be relaxed. This is still an ordinary store, unless it was volatile in IL. We just need to emit it as release, but can do all kind of optimizations. It might need a new flag.

Reply:
Nit: If we need to emit it as release, it is not really an ordinary store.

Reply:
What I meant - if it was a regular store in IL, we can still move it around or coalesce with other writes to same location - for optimization purposes it is still a regular store. I think `GTF_IND_VOLATILE` would suppress all that, but it could be ok for now.

Reply:
Yeah, thanks, by the time we reach this phase we already have done all the transformations/optimizations so `GTF_IND_VOLATILE` should not really cause codegen diffs apart from the `str` -> `stlr` on arm

Reply:
![image](https://user-images.githubusercontent.com/523221/194428329-8094133c-6c4d-4ff8-b6d9-f1f0522c7953.png)

Seems to work, here jit decided to use `str`+`dmb` instead of `stlr` because of addressing mode